packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -347,6 +355,17 @@ export class RichTextEditor extends FoundationElement {
                 return false;
         }
     }
+
+    // In Firefox browser, Once the editor gets focused, the blinking caret will be visible until we click format buttons (Bold, Italic ...) in the Firefox browser (Changing focus).
+    // But once any of the toolbar button is clicked, editor internally has its focus but the blinking caret disappears.
+    // As a workaround, manually triggering blur and setting focus on editor makes the blinking caret to re-appear.
+    private focusEditorInFirefox(): void {
+        const browserInfo = navigator.userAgent;
+        if (browserInfo.includes('Firefox')) {
```

Comment:
User agent detection is [notoriously fickle](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#browser_name_and_version). Browsers often lie about their user agent and you have to be careful about detecting the browser name (Firefox) vs the rendering agent (Mozilla). Is this code safe to run on all browsers? If so then I'd vote to remove the user agent check and run it unconditionally. (We could rename the function to something like `forceFocusEditor()`if we do this).

Reply:
Thanks for relieving the condition check here! I was also skeptical on this implementation and was exploring whether any other recommended ways are possible. 

Since this was a browser specific bug and not a feature we were also not able to use [feature detection](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection). Just curious to hear if there are any other good ways of identifying a browser, however these situations are rare.

Reply:
I'm not aware of strategies other than the ones discussed in that MDN article.